subgraphs/rocket-pool/src/entityUpdates/financialMetrics.ts

```diff
@@ -187,10 +182,22 @@ export function updateTotalRevenueMetrics(
     block
   );
 
-  // Staking Rewards
+  let additionalRewards = BIGDECIMAL_ZERO;
+
+  const lastPriceUsd = getOrCreateToken(
+    Address.fromString(ETH_ADDRESS),
+    block.number
+  ).lastPriceUSD!;
+
+  const currentEthAmount = bigIntToBigDecimal(pool.inputTokenBalances[1]);
 
+  if (stakingRewards > currentEthAmount) {
+    additionalRewards = stakingRewards.minus(currentEthAmount);
+  }
+
+  const ethAmount = currentEthAmount.plus(additionalRewards);
   // Pool
-  pool.cumulativeTotalRevenueUSD = stakingRewards;
+  pool.cumulativeTotalRevenueUSD = ethAmount.times(lastPriceUsd);
```

Comment:
Hmm 🤔  
If the price of ETH changes over time, cumultiveRevenueUSD will go up and down.

Because it is a cumulative value it should only go up. So instead you should add `additionalRewards * ETHPrice` to the cumulative. Same as you are doing for the daily and hourly cumulative revenues.